docs/reference/how-to/knn-search.asciidoc

```diff
@@ -0,0 +1,98 @@
+[[tune-knn-search]]
+== Tune approximate kNN search
+
+{es} supports <<approximate-knn, approximate k-nearest neighbor search>> for
+efficiently finding the _k_ nearest vectors to a query vector. Since
+approximate kNN search works differently from other queries, there are special
+considerations around its performance.
+
+Many of these recommendations help improve search speed. With approximate kNN,
+the indexing algorithm runs searches under the hood to create the vector index
+structures. So these same recommendations also help with indexing speed.
+
+[discrete]
+=== Ensure data nodes have enough memory
+
+{es} currently uses the https://arxiv.org/abs/1603.09320[HNSW] algorithm for
+approximate kNN search. HNSW is a graph-based algorithm which only works
+efficiently when all vector data is held in memory. You should ensure that
+data nodes have enough RAM to hold the vector data and index structures. To
+check the size of the vector data, you can use the <<indices-disk-usage>> API.
```

Comment:
Just to confirm, we don't use any compression when storing vector data on disk?

Reply:
Should we provide an approximate formulate for the lower threshold for RAM: `N of docs * dims * 4` for vectors + `N of docs * M * 2` for graph?

Reply:
Could you explain how compression is related here? And sounds good, I can add an approximate calculation here.

Reply:
> Could you explain how compression is related here?

Sorry for a confusion, we are not using any disk compression for the approximate knn vector values.  We compress only binary doc values for the exact search.

Reply:
I pushed an update to improve this section.